### PR TITLE
[sitecore-jss-dev-tools] Update sitecore packages metadata generation logic to account for monorepos

### PR DESCRIPTION
**Pull request by yavorsk**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation

Previously getMetadata() was generating the metadata about the used sitecore packages through traversing the node-modules folder of the application. This approach does not work in a monorepo setup.
This PR addresses the problem by making some updates of the logic that gathers the metadata - it uses the `npm query` command to gather the information of the used sitecore packages and their exact version.

## Testing Details

- [x] Unit Test Added
- [x] Manual Test/Other (Please elaborate) Test standalone, in monorepo, in other monorepos

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by art-alexeyenko — CHANGES_REQUESTED** [review]
Would it also make sense to cleanup some of the test-data package lock files for better maintainability?

File: packages/sitecore-jss-dev-tools/src/templating/metadata.ts [hunk 1]
```diff
@@ -1,31 +1,90 @@
...
+    } catch (error) {
+      console.error('Failed to create package-lock.json in project', error);
+      return metadata;
     }
-  });
+  }
+
+  let packageLock = {};
+  try {
+    packageLock = JSON.parse(fs.readFileSync(packageLockPath, 'utf8'));
```
[the remark this answers is not in the document]
Comment (yavorsk) [reply packages/sitecore-jss-dev-tools/src/templating/metadata.ts:35]: in my tests, when it's generated with `--workspaces true` package-lock gets created in the root of the monorepo so app won't find it and generate its own.

File: packages/sitecore-jss-dev-tools/src/templating/metadata.ts [hunk 2]
```diff
@@ -1,31 +1,90 @@
...
-          metadata.packages[json.name] = json.version;
-        } catch (e) {
-          console.error(`Failed to read/parse package.json for ${pkg}`, e);
-        }
-      });
+
+  const packageLockPath = './package-lock.json';
+  const packageLockIsAvailable = fs.existsSync(packageLockPath);
+
+  if (!packageLockIsAvailable) {
```
[the remark this answers is not in the document]
Comment (yavorsk) [reply packages/sitecore-jss-dev-tools/src/templating/metadata.ts:24]: The problem is that going through node_modules can produce a result and we would not be certain if it is correct in monorepo setup..

File: packages/sitecore-jss-dev-tools/src/templating/metadata.ts [hunk 3]
```diff
@@ -1,31 +1,91 @@
...
-import { Metadata } from '@sitecore-jss/sitecore-jss/editing';
+import { execSync } from 'child_process';
+
+type Package = {
+  [key: string]: unknown;
+  name: string;
+  version: string;
+};
+
+const dependencyTypes = ['dependencies', 'devDependencies', 'peerDependencies'];
```
[the remark this answers is not in the document]
Comment (illiakovalenko) [reply packages/sitecore-jss-dev-tools/src/templating/metadata.ts]: Not relevant anymore

File: CHANGELOG.md [hunk 4]
```diff
@@ -67,6 +67,7 @@ Our versioning strategy is as follows:
 * `[template/node-xmcloud-proxy]` `[sitecore-jss-proxy]` Introduced /api/healthz endpoint ([#1928](https://github.com/Sitecore/jss/pull/1928))
 * `[sitecore-jss]` `[sitecore-jss-angular]` Render field metdata chromes in editMode metadata - in edit mode metadata in Pages, angular package field directives will render wrapping `code` elements with field metadata required for editing; ([#1926](https://github.com/Sitecore/jss/pull/1926))
 * `[sitecore-jss-nextjs]` Expose MiddlewareBase class and MiddlewareBaseConfig type ([#1941](https://github.com/Sitecore/jss/pull/1941))
+* `[sitecore-jss-dev-tools]` getMetadata() now uses `npm query` command to get the list and exact versions of packages. this solution works for monorepo setups ([#1949](https://github.com/Sitecore/jss/pull/1949))
```
[the remark this answers is not in the document]
Comment (yavorsk) [reply CHANGELOG.md:70]: yep makes sense

**Review by art-alexeyenko — APPROVED** [review]

**Pull request merged**